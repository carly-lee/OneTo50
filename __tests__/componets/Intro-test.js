import 'react-native';
import React from 'react';
import Intro from 'components/intro';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe( 'Intro', ()=>{
	const navigation = { navigate: jest.fn() };

  it( 'renders correctly',()=>{
    const tree = renderer.create( <Intro navigation={ navigation } /> );
    expect( tree ).toMatchSnapshot();
  });

  it( 'calls onPress() when a user presses the button', ()=>{
    const wrapper = shallow( <Intro navigation={ navigation } /> );
    wrapper.find( 'Button' ).simulate( 'press' );
    expect( navigation.navigate ).toBeCalled();
  });
});
