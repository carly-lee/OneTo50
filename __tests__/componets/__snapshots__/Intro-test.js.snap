// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 48,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    1 to 50
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel="Let's Start!"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }
      >
        Let's Start!
      </Text>
    </View>
  </View>
</View>
`;
